docs/examples/deployment-rollout/README.md

```diff
@@ -0,0 +1,46 @@
+# Rollout Example
+Here is an example of how to rollout an application with a component of type deployment. 
+
+
+## Rollout steps
+1. Install deployment based workloadDefinition
+```shell
+kubectl apply -f docs/examples/deployment-rollout/webservice-definition.yaml
+```
+
+2. Apply an application 
+```shell
+kubectl apply -f docs/examples/deployment-rollout/app-source.yaml
+```
+Wait for the application's status to be "running"
+
+3. Prepare the application for rolling out
+```shell
+kubectl apply -f docs/examples/deployment-rollout/app-source-prep.yaml
+```
+Wait for the applicationConfiguration "test-rolling-v1" `Rolling Status` to be "RollingTemplated"
+
+4. Modify the application image and apply
+```shell
+kubectl apply -f docs/examples/deployment-rollout/app-target.yaml
+```
+Wait for the applicationConfiguration "test-rolling-v2" `Rolling Status` to be "RollingTemplated"
```

Comment:
as we discussed,  these two steps need to be merged.

Reply:
step 3 and 4?

Reply:
yes, right？

Reply:
step 3 is to mark source is going to be rollout and step 4 is to modify the target. I am not sure how to combine these two. 

We can probably remove the step 5